core/src/main/java/tc/oc/pgm/teams/TeamMatchModule.java

```diff
@@ -573,9 +612,32 @@ public boolean kickPlayerOffTeam(Team kickFrom, boolean forBalance) {
 
   @EventHandler(priority = EventPriority.MONITOR)
   public void onPartyChange(PlayerPartyChangeEvent event) {
+    MatchPlayer player = event.getPlayer();
+
     if (event.getNewParty() instanceof Team
         || (event.getNewParty() instanceof ObserverParty && event.getOldParty() != null)) {
-      event.getPlayer().sendMessage(translatable("join.ok.team", event.getNewParty().getName()));
+      Component joinMsg = translatable("join.ok.team", event.getNewParty().getName());
+      boolean title = false;
+
+      // Players with team switch kit, check for title value and remove from map
+      if (hasTeamSwitchKit(player)) {
+        title = showTeamSwitchTitle(player);
+        setTeamSwitchKit(player, false, false);
+      }
+
+      if (title) {
```

Comment:
1.7 players can't see titles, they need to keep the chat messages

Reply:
Oh good catch, I'll add in a legacy check